### PR TITLE
Np 1384 include username

### PR DESCRIPTION
**Pull request by axthosarouris**
Changes:

1. Re-use constants and logic for reading a username from RequestInfo
2. Adjust FindDoiRequestsHandler
3. CreateDoiRequestHandler reads username from RequestInfo and sends it to the DoiRequestService
4. DoiRequestService validates and adds the username to the DoiRequest

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,15 +25,23 @@ subprojects {
     // source-code version and must be <= targetCompatibility
     targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 
-    def junit5Version = '5.6.0'
+    ext{
+        jacksonVersion = '2.10.2'
```
Comment (axthosarouris) [inline build.gradle:29]: we cannot define global variables but we can define extra properties for libraries that are used in multiple projects

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 2]
```diff
@@ -120,28 +122,23 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
             .withDate(Instant.now(clockForTimestamps))
-            .addMessage(createMessage(createDoiRequest))
+            .addMessage(createMessage(createDoiRequest, username))
             .build();
+
         publication.setDoiRequest(doiRequest);
         putItem(publication);
     }
 
-    private DoiRequestMessage createMessage(CreateDoiRequest createDoiRequest) {
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:137]: moved lower

File: create-doi-request/src/main/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandler.java [hunk 3]
```diff
@@ -51,7 +52,8 @@ private static Logger defaultLogger() {
     protected Void processInput(CreateDoiRequest input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
         input.validate();
-        doiRequestService.createDoiRequest(input);
+        String username = UserDetails.getUsername(requestInfo);
```
Comment (axthosarouris) [inline create-doi-request/src/main/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandler.java:55]: This is where the handler supplies the username to the service.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 4]
```diff
@@ -152,6 +149,21 @@ private DoiRequestMessage createMessage(CreateDoiRequest createDoiRequest) {
         return Optional.ofNullable(doiRequestSummary);
     }
 
+    private void validateUsername(Publication publication, String username) throws ForbiddenException {
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:152]: new method for implementing the check on the username

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/userdetails/UserDetails.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package no.unit.nva.doi.requests.userdetails;
+
+import static nva.commons.utils.RequestUtils.getRequestContextParameter;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import nva.commons.handlers.RequestInfo;
+
+public final class UserDetails {
+
+    public static final String ROLE = "role";
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/userdetails/UserDetails.java:10]: Functionality taken out from `FindDoiRequestHandler` to re-use in `CreateRequestHandler`.

File: doi-requests-commons/src/test/java/no/unit/nva/doi/requests/userdetails/UserDetailsTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+public class UserDetailsTest {
+
+    private static final String FEIDE_ID_IN_FILE = "feide@id";
+    private static final String CUSTOMER_ID_IN_FILE = "123456";
+    private static final String ROLES_IN_FILE = "Role1,Role2,Role3";
+    private RequestInfo requestInfo;
+
+    @BeforeEach
+    public void init() throws JsonProcessingException {
```
Comment (axthosarouris) [inline doi-requests-commons/src/test/java/no/unit/nva/doi/requests/userdetails/UserDetailsTest.java:23]: Tests for code coverage. Bad case scenarios were not implemented as the  methods perform only a library call.

File: doi-requests-commons/src/test/resources/requestInfoWithContext.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "requestContext": {
```
Comment (axthosarouris) [inline doi-requests-commons/src/test/resources/requestInfoWithContext.json:2]: expected format of the user-info in the requestContext

File: find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java [hunk 8]
```diff
@@ -74,10 +70,10 @@ protected DoiRequestsResponse processInput(Void input, RequestInfo requestInfo,
         String assignedRoles;
         String customerId;
         try {
-            user = getRequestContextParameter(requestInfo, FEIDE_ID);
+            user = UserDetails.getUsername(requestInfo);
```
Comment (axthosarouris) [inline find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java:73]: Code replaced by the common `UserDetails` class

File: build.gradle [hunk 9]
```diff
@@ -25,15 +25,23 @@ subprojects {
...
+
 
     dependencies {
         testImplementation group: 'com.github.BIBSYSDEV', name: 'nva-testutils', version: '0.1.8'
         testImplementation group: 'org.hamcrest', name: 'hamcrest', version: '2.2'
         testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.2.4'
-        testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: junit5Version
-        testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-migrationsupport', version: junit5Version
-        testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: junit5Version
+        testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: project.ext.get("junit5Version")
```
Comment (brinxmat) [inline build.gradle:42]: can you not use the project.junit5Version syntax?

File: build.gradle [hunk 10]
```diff
@@ -25,15 +25,23 @@ subprojects {
     // source-code version and must be <= targetCompatibility
     targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 
-    def junit5Version = '5.6.0'
+    ext{
```
Comment (brinxmat) [inline build.gradle:28]: project.ext, I think

File: create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java [hunk 11]
```diff
@@ -116,22 +129,53 @@ public void handleRequestSavesDoiRequestToPublicationWhenPublicationIdIsNotEmpty
...
     }
 
-    private <T> GatewayResponse<T> sendRequest(CreateDoiRequest doiRequest) throws IOException {
-        InputStream input = createRequest(doiRequest);
+    private void assertThatLogsContainReasonForForbiddenMessage(TestAppender appender, Publication publication) {
+        String expectedErrorMessage = String.format(WRONG_OWNER_ERROR, INVALID_USERNAME, publication.getOwner());
+        assertThat(appender.getMessages(), containsString(expectedErrorMessage));
+    }
+
+    private void assertThatProblemDetailsDoesRevealSensitiveInformation(Problem details,
```
Comment (brinxmat) [inline create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java:169]: Nice!

File: create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java [hunk 12]
```diff
@@ -173,10 +217,19 @@ private CreateDoiRequest requestWithoutPublicationId() {
         return new CreateDoiRequest();
     }
 
-    private InputStream createRequest(CreateDoiRequest doiRequest)
+    private InputStream createRequest(CreateDoiRequest doiRequest, String username)
         throws com.fasterxml.jackson.core.JsonProcessingException {
-        return new HandlerRequestBuilder<CreateDoiRequest>(JsonUtils.objectMapper)
+        ObjectNode requestContext = objectMapper.createObjectNode();
+        requestContext
+            .putObject("authorizer")
```
Comment (brinxmat) [inline create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java:224]: These can be constants

File: doi-request-common-constants/build.gradle [hunk 13]
```diff
@@ -0,0 +1,3 @@
+dependencies {
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: project.ext.get("jacksonVersion")
```
Comment (brinxmat) [inline doi-request-common-constants/build.gradle:2]: Use dot syntax

File: doi-requests-commons/build.gradle [hunk 14]
```diff
@@ -3,29 +3,27 @@ repositories {
     maven { url "https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/release" }
 }
 
-def jacksonVersion = '2.10.2'
-def awsSdkVersion = '1.11.714'
 
 dependencies {
-    api group: 'com.github.BIBSYSDEV', name: 'nva-commons', version: '78adbd6fe3'
-    api group: 'com.github.BIBSYSDEV', name: 'nva-datamodel-java', version: '043c5d676211'
+    api group: 'com.github.BIBSYSDEV', name: 'nva-commons', version: project.ext.get("nvaCommonsVersion")
```
Comment (brinxmat) [inline doi-requests-commons/build.gradle:8]: Use dot syntax

File: test-module/build.gradle [hunk 15]
```diff
@@ -3,17 +3,14 @@ repositories {
     maven { url "https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/release" }
 }
 
-def awsSdkVersion = '1.11.714'
-def junit5Version = '5.6.0'
-
 dependencies {
     implementation( project (":doi-request-common-constants"))
-    implementation group: 'com.github.BIBSYSDEV', name: 'nva-datamodel-java', version: '043c5d676211'
+    implementation group: 'com.github.BIBSYSDEV', name: 'nva-datamodel-java', version: project.ext.get("nvaDatamodelJavaVersion")
```
Comment (brinxmat) [inline test-module/build.gradle:8]: Dot syntax

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 16]
```diff
@@ -25,23 +25,22 @@ subprojects {
...
     targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 
-    ext{
+    project.ext{
         jacksonVersion = '2.10.2'
         awsSdkVersion = '1.11.714'
         junit5Version = '5.6.0'
-        nvaCommonsVersion = '78adbd6fe3'
-        nvaDatamodelJavaVersion = '043c5d676211'
+        nvaCommonsVersion = '0.4.4'
```
Comment (brinxmat) [inline build.gradle:32]: Consider also bumping aws, junit, mockito to latest, or setting up Dependabot

**Review by brinxmat — APPROVED** [review]

**Pull request merged**